modules/product/src/main/java/com/opengamma/strata/product/swap/type/InflationRateSwapLegConvention.java

```diff
@@ -0,0 +1,644 @@
+   */
+  @PropertyDefinition(get = "field")
+  private final DayCount dayCount;
+  
+  /**
+   * The flag indicating whether to exchange the notional.
+   * <p>
+   * If 'true', the notional there is both an initial exchange and a final exchange of notional.
+   * <p>
+   * This will default to 'false' if not specified.
+   */
+  @PropertyDefinition
+  private final boolean notionalExchange;
+
+  //-------------------------------------------------------------------------
+  /**
+   * Obtains a convention based on the specified index.
+   * <p>
+   * The standard market convention for an Inflation rate leg is based on the index
+   * Use the {@linkplain #builder() builder} for unusual conventions.
+   * 
+   * @param index  the index, the market convention values are extracted from the index
+   * @return the convention
+   */
+  public static InflationRateSwapLegConvention of(PriceIndex index) {
+    return InflationRateSwapLegConvention.builder()
+        .index(index)
+        .build();
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Gets the leg currency, optional with defaulting getter.
+   * <p>
+   * This is the currency of the swap leg and the currency that payment is made in.
+   * The data model permits this currency to differ from that of the index,
+   * however the two are typically the same.
+   * <p>
+   * This will default to the currency of the index if not specified.
+   * 
+   * @return the start date business day adjustment, not null
+   */
+  public Currency getCurrency() {
+    return currency != null ? currency : index.getCurrency();
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Creates a leg based on this convention.
+   * <p>
+   * This returns a leg based on the specified date.
+   * The notional is unsigned, with pay/receive determining the direction of the leg.
+   * If the leg is 'Pay', the fixed rate is paid to the counterparty.
+   * If the leg is 'Receive', the fixed rate is received from the counterparty.
+   *
+   * @param startDate  the start date
+   * @param endDate  the end date
+   * @param payReceive  determines if the leg is to be paid or received
+   * @param lag       the positive period between the price index and the accrual date, typically a number of months
+   * @param businessDayAdjustment the business day adjustment to apply
+   * @param paymentDateOffset an adjustment that alters the payment date by adding a period of days
+   * @param notional  the notional
+   * @return the leg
+   */
+  public RateCalculationSwapLeg toLeg(
+      LocalDate startDate,
+      LocalDate endDate,
+      PayReceive payReceive,
+      Period lag,
+      BusinessDayAdjustment businessDayAdjustment,
+      DaysAdjustment paymentDateOffset,
+      double notional) {
+
+    return toLeg(startDate, endDate, payReceive, lag, businessDayAdjustment, paymentDateOffset, notional, false);
```

Comment:
The interpolated/monthly flag should be added in the convention fields. Not hard-coded here.